src/lib.rs

```diff
@@ -665,13 +665,24 @@ pub enum ConfigSetting {
 ///
 /// If CONFIG_KCONFIG_PROC is available, the config is read from `/proc/config.gz`.
 /// Else look in `/boot/config-$(uname -r)` or `/boot/config` (in that order).
+///
+/// # Notes
+/// Reading the compress `/proc/config.gz` is only supported if the `flate2` feature is enabled
+/// (which it is by default).
+#[cfg_attr(feature = "flate2", doc = "The flate2 feature is currently enabled")]
+#[cfg_attr(not(feature = "flate2"), doc = "The flate2 feature is NOT currently enabled")]
 pub fn kernel_config() -> ProcResult<HashMap<String, ConfigSetting>> {
-    use flate2::read::GzDecoder;
-
-    let reader: Box<dyn BufRead> = if Path::new(PROC_CONFIG_GZ).exists() {
-        let file = FileWrapper::open(PROC_CONFIG_GZ)?;
-        let decoder = GzDecoder::new(file);
-        Box::new(BufReader::new(decoder))
+    let reader: Box<dyn BufRead> = if Path::new(PROC_CONFIG_GZ).exists() && cfg!(feature = "flate2") {
```

Comment:
I think `&& cfg!(feature = "flate2")` is useless

Reply:
if cfg!(feature = "flate2") == false,  block would not execute

if cfg!(feature = "flate2") == true, block would execute `flate2::read::GzDecoder`

so the cfg(not(...flate2 block may never execute

Reply:
if cfg!(feature = "flate2") == false,  block would not execute

if cfg!(feature = "flate2") == true, block would execute `flate2::read::GzDecoder`

so the cfg(not(...flate2 block may never execute

Reply:
Yes, the `#[cfg(not(feature = "flate2"))]` block is unreachable, but is needed to typecheck